system/src/Grav/Common/Grav.php

```diff
@@ -520,6 +350,96 @@ public function shutdown()
     }
 
     /**
+     * Magic Catch All Function
+     * Used to call closures like measureTime on the instance.
+     * Source: http://stackoverflow.com/questions/419804/closures-as-class-members
+     */
+    public function __call($method, $args)
+    {
+        $closure = $this->$method;
+        call_user_func_array($closure, $args);
+    }
+
+    /**
+     * Initialize and return a Grav instance
+     *
+     * @param  array $values
+     *
+     * @return static
+     */
+    protected static function load(array $values)
+    {
+        $container = new static($values);
+
+        $container['grav'] = $container;
+
+        $container['debugger'] = new Debugger();
+        $debugger = $container['debugger'];
+
+        // closure that measures time by wrapping a function into startTimer and stopTimer
+        // The debugger can be passed to the closure. Should be more performant
+        // then to get it from the container all time.
+        $container->measureTime = function($timerId, $timerTitle, $callback) use ($debugger) {
+          $debugger->startTimer($timerId, $timerTitle);
+          $callback($debugger);
```

Comment:
I do not like having $debugger as a parameter. The parameter seems to have no purpose, which is why I ended up using a lot of time trying to understand reasoning behind it. I see it being used in 2 processors out of 13, but then again, its already available through the container. If its just there because of optimization, there are better ways to optimize code.